pkg/cli/cmd/app/delete/delete.go

```diff
@@ -160,38 +164,48 @@ func (r *Runner) Run(ctx context.Context) error {
 		return err
 	}
 
-	app, err := client.GetApplication(ctx, r.ApplicationName)
-	if clients.Is404Error(err) {
-		r.Output.LogInfo("Application '%s' does not exist or has already been deleted.", r.ApplicationName)
-		return nil
-	} else if err != nil {
-		return err
-	}
+	if !r.Confirm {
```

Comment:
The confirmation logic now only runs when `!r.Confirm`, but the environment name extraction is also inside this block. This means when `--yes` is used, `r.EnvironmentName` won't be set, which could cause issues with the progress text.